core/domain/base_model_validators.py

```diff
@@ -62,14 +64,18 @@
 ERROR_CATEGORY_TYPE_CHECK = 'type check'
 ERROR_CATEGORY_VERSION_CHECK = 'version check'
 ERROR_CATEGORY_STALE_CHECK = 'stale check'
+ERROR_CATEGORY_INVALID_USER_SETTING_IDS = 'invalid user setting ids'
 
 VALIDATION_MODE_NEUTRAL = 'neutral'
 VALIDATION_MODE_STRICT = 'strict'
 VALIDATION_MODE_NON_STRICT = 'non-strict'
 
 
 class ExternalModelFetcherDetails(python_utils.OBJECT):
-    """Value object providing the class and ids to fetch an external model."""
+    """Value object providing the class and ids to fetch an external model.
+    NOTE TO DEVELOPERS: For UserSettingsModel, use the
+    UserSettingsModelFetcherDetails class instead of this one.
+    """
 
     def __init__(
             self, field_name, model_class, model_ids):
```

Comment:
nit: while we are here
```suggestion
    def __init__(self, field_name, model_class, model_ids):
```

Reply:
Done.